### ISSUE TITLE
6.0 GA readiness

### ISSUE THREAD

**Issue by Mpdreamz** [elastic/elasticsearch-net#2866]
# 6.0 readiness

This issue tracks all of the outstanding issues we need to fix before we can release the 6.0 version of the client.

## Failing integration collections 

These are uncategorized right now but will be updated with the cause/issue/pr while going over it.

 - [ ] ReadOnly: Tests.Indices.IndexManagement.TypesExists.TypeExistsApiTests
 - [ ] ReadOnly: Tests.Aggregations.Metric.TopHits.TopHitsAggregationUsageTests
 - [ ] ReadOnly: Tests.Aggregations.Bucket.Filter.EmptyFilterAggregationUsageTests
 - [ ] ReadOnly: Tests.Search.Request.FielddataFieldsUsageTests
 - [ ] ReadOnly: Tests.Search.MultiSearch.MultiSearchTemplate.MultiSearchTemplateApiTests
 - [ ] ReadOnly: Tests.ClientConcepts.Troubleshooting.DeprecationLogging
 - [ ] ReadOnly: Tests.Indices.MappingManagement.GetMapping.GetMappingNonExistentIndexApiTests
 - [ ] ReadOnly: Tests.Indices.StatusManagement.ClearCache.ClearCacheApiTests
 - [ ] ReadOnly: Tests.Aggregations.Bucket.Children.ChildrenAggregationUsageTests
 - [ ] ReadOnly: Tests.Aggregations.ChildrenAggregationFluentAggsUsageTests
 - [ ] ReadOnly: Tests.Search.FieldCapabilities.FieldCapabilitiesApiTests
 - [ ] ReadOnly: Tests.Search.Request.HighlightingUsageTests
 - [ ] ReadOnly: Tests.Document.Multiple.MultiGet.MultiGetParentApiTests
 - [ ] ReadOnly: Tests.Cluster.ClusterState.ClusterStateApiTests
 - [ ] ReadOnly: Tests.Search.MultiSearch.MultiSearchApiTests
 - [ ] Writable: Tests.Document.Single.Create.CreateJObjectIntegrationTests
 - [ ] Writable: Tests.Document.Single.Index.IndexIntegrationTests
 - [ ] Writable: Tests.Document.Single.Index.IndexAnonymousTypesIntegrationTests
 - [ ] Writable: Tests.Document.Single.Delete.DeleteApiTests
 - [ ] Writable: Tests.Document.Single.Create.CreateAnonymousTypesIntegrationTests
 - [ ] Writable: Tests.QueryDsl.Specialized.Percolate.PercolateQueryExistingDocumentUsageTests
 - [ ] Writable: Tests.Document.Single.Create.CreateInvalidApiTests
 - [ ] Writable: Tests.Document.Single.Index.IndexJObjectIntegrationTests
 - [ ] Writable: Tests.Reproduce.GithubIssue2788
 - [ ] Writable: Tests.QueryDsl.Specialized.Percolate.PercolateQueryUsageTests
 - [ ] XPack: Tests.XPack.Security.User.InvalidateUserAccessToken.InvalidateUserAccessTokenApiTests
 - [ ] XPack: Tests.XPack.Security.User.InvalidateUserAccessToken.InvalidateUserAccessTokenBadPasswordApiTests
 - [ ] XPack: Tests.XPack.Security.User.GetUserAccessToken.GetUserAccessTokenApiTests
 - [ ] IntrusiveOperation: Tests.Search.Request.QueryInnerHitsApiTests
 - [ ] IntrusiveOperation: Tests.Modules.Scripting.ScriptingCrudTests
 - [ ] ManualReindex: Tests.Document.Multiple.Reindex.ReindexApiTests

## New Features

Not extensive, need a final pass over release notes and core github issues.

- [ ] Forward port of the ML API's #2856 
- [ ] Grok processor
- [ ] Nodes Usage API
- [ ] XPack deprecation info API
- [ ] Join data type mapping
- [ ] Index sorting

## Breaking changes

Go over https://www.elastic.co/guide/en/elasticsearch/reference/6.x/breaking-changes-6.0.html and list each PR here.

**Comment by Mpdreamz** [comment]
Closing this in favor of tracking on the PR 

https://github.com/elastic/elasticsearch-net/pull/2870

**Comment by russcam** [comment]
I forward ported ML APIs but they will need a pass over in case any of the APIs have changed from 5.x
